data_structures/src/chain.rs

```diff
@@ -1939,7 +1939,102 @@ pub enum DataRequestStage {
+    ///   (how many blocks were consolidated before this one).
+    /// * A reference count, used to keep this map clear after removing transactions.
+    ///   This reference count is the number of output pointers that point to this
+    ///   transaction hash, when it reaches 0 the entry should be removed.
+    transaction_block_number: HashMap<Hash, (u32, u32)>,
+}
+
+impl UnspentOutputsPool {
+    pub fn get<Q: ?Sized>(&self, k: &Q) -> Option<&ValueTransferOutput>
+    where
+        OutputPointer: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq,
+    {
+        self.map.get(k)
+    }
+
+    pub fn contains_key<Q: ?Sized>(&self, k: &Q) -> bool
+    where
+        OutputPointer: std::borrow::Borrow<Q>,
+        Q: std::hash::Hash + Eq,
+    {
+        self.map.contains_key(k)
+    }
+
+    pub fn insert(
+        &mut self,
+        k: OutputPointer,
+        v: ValueTransferOutput,
+        block_number: u32,
+    ) -> Option<ValueTransferOutput> {
+        let transaction_id = k.transaction_id;
+        let old_vto = self.map.insert(k, v);
+        if old_vto.is_none() {
+            // Store block number for this transaction hash
+            // If the transaction hash already exists, do not update the block number
+            let (_block_number, refcount) = self
+                .transaction_block_number
+                .entry(transaction_id)
+                .or_insert((block_number, 0));
+            // Increase the refcount if this is a new output pointer
+            // Entries should be removed from the map when the refcount reaches 0 again
+            *refcount += 1;
+        }
+
+        old_vto
+    }
+
+    pub fn remove(&mut self, k: &OutputPointer) -> Option<ValueTransferOutput> {
+        let vto = self.map.remove(k);
+
+        if vto.is_some() {
+            // Decrease refcount if this is a new output pointer
+            let (_epoch, refcount) = self
+                .transaction_block_number
+                .get_mut(&k.transaction_id)
+                .unwrap();
+            *refcount -= 1;
+            // Entries should be removed from the map when the refcount reaches 0
+            if *refcount == 0 {
+                self.transaction_block_number.remove(&k.transaction_id);
+            }
+        }
+
+        vto
+    }
+
+    pub fn drain(
+        &mut self,
+    ) -> std::collections::hash_map::Drain<OutputPointer, ValueTransferOutput> {
+        self.map.drain()
+    }
+
+    pub fn iter(&self) -> std::collections::hash_map::Iter<OutputPointer, ValueTransferOutput> {
+        self.map.iter()
+    }
+
+    /// Returns the number of the block that included the transaction referenced
+    /// by this OutputPointer. The difference between that number and the
+    /// current number of consolidated blocks is the "coin age".
+    pub fn included_in_block_number(&self, k: &OutputPointer) -> Option<Epoch> {
+        if !self.map.contains_key(k) {
```

Comment:
Please honor [rule #8](https://www.robertsrulesofcoders.com/)

Reply:
The only rules I follow are the clippy lints

Reply:
Lol. Did it complain about this?

Reply:
No, and I think that lint does not exist. Should be easy to implement, maybe you can open them an issue :laughing: